packages/@aws-cdk/aws-ec2/lib/client-vpn-endpoint.ts

```diff
@@ -0,0 +1,385 @@
+    }
+    return new Import(scope, id);
+  }
+
+  public readonly endpointId: string;
+
+  /**
+   * Allows specify security group connections for the endpoint.
+   */
+  public readonly connections: Connections;
+
+  public readonly targetNetworksAssociated: IDependable;
+
+  private readonly _targetNetworksAssociated = new ConcreteDependable();
+
+  constructor(scope: Construct, id: string, props: ClientVpnEndpointProps) {
+    super(scope, id);
+
+    if (!Token.isUnresolved(props.vpc.vpcCidrBlock)) {
+      const clientCidr = new CidrBlock(props.cidr);
+      const vpcCidr = new CidrBlock(props.vpc.vpcCidrBlock);
+      if (vpcCidr.containsCidr(clientCidr)) {
+        throw new Error('The client CIDR cannot overlap with the local CIDR of the VPC');
+      }
+    }
+
+    if (props.dnsServers && props.dnsServers.length > 2) {
+      throw new Error('A client VPN endpoint can have up to two DNS servers');
+    }
+
+    if (props.logging == false && (props.logGroup || props.logStream)) {
+      throw new Error('Cannot specify `logGroup` or `logStream` when logging is disabled');
+    }
+
+    const logging = props.logging ?? true;
+    const logGroup = logging
+      ? props.logGroup ?? new logs.LogGroup(this, 'LogGroup')
+      : undefined;
+
+    const securityGroups = props.securityGroups ?? [new SecurityGroup(this, 'SecurityGroup', {
+      vpc: props.vpc,
+    })];
+    this.connections = new Connections({ securityGroups });
+
+    const endpoint = new CfnClientVpnEndpoint(this, 'Resource', {
+      authenticationOptions: renderAuthenticationOptions(props.clientCertificate, props.userBasedAuthentication),
+      clientCidrBlock: props.cidr,
+      clientConnectOptions: props.clientConnectionHandler
+        ? {
+          enabled: true,
+          lambdaFunctionArn: props.clientConnectionHandler.functionArn,
+        }
+        : undefined,
+      connectionLogOptions: {
+        enabled: logging,
+        cloudwatchLogGroup: logGroup?.logGroupName,
+        cloudwatchLogStream: props.logStream?.logStreamName,
+      },
+      description: props.description,
+      dnsServers: props.dnsServers,
+      securityGroupIds: securityGroups.map(s => s.securityGroupId),
+      selfServicePortal: booleanToEnabledDisabled(props.selfServicePortal),
+      serverCertificateArn: props.serverCertificate.certificateArn,
+      splitTunnel: props.splitTunnel,
+      transportProtocol: props.transportProtocol,
+      vpcId: props.vpc.vpcId,
+      vpnPort: props.port,
+    });
+
+    this.endpointId = endpoint.ref;
+
+    if (props.userBasedAuthentication && (props.selfServicePortal ?? true)) {
+      // Output self-service portal URL
+      new CfnOutput(this, 'SelfServicePortalUrl', {
+        value: `https://self-service.clientvpn.amazonaws.com/endpoints/${this.endpointId}`,
+      });
+    }
+
+    // Associate subnets
+    const subnetIds = props.vpc.selectSubnets(props.vpcSubnets).subnetIds;
```

Comment:
`subnetIds` might be a list-encoded token (people like to import VPCs from parameters or Exports containing lists of subnet IDs), in which case the iteration below cannot work.

Detect that situation (`Token.isUnresolved()`) and throw an appropriate error.